pkg/R/DataFrame.R

```diff
@@ -114,20 +114,189 @@ setMethod("count",
             callJMethod(sdf, "count")
           })
 
-#' Collect elements of a DataFrame
-#' 
-#' Returns a list of Row objects from a DataFrame
-#' 
-#' @param df A SparkSQL DataFrame
-#' 
+# Collect the rows of a Spark DataFrame and return a named list for each row.
+
 #' @rdname collect-methods
 #' @export
+#' @examples
+#'\dontrun{
+#' sc <- sparkR.init()
+#' sqlCtx <- sparkRSQL.init(sc)
+#' path <- "path/to/file.json"
+#' df <- jsonFile(sqlCtx, path)
+#' collected <- collect(df)
+#' firstName <- collected[[1]]$name
+#' }
 
-# TODO: Collect() currently returns a list of Generic Row objects and is WIP.  This will eventually 
-# be part of the process to read a DataFrame into R and create a data.frame.
 setMethod("collect",
           signature(rdd = "DataFrame"),
-          function(rdd){
-            sdf <- rdd@sdf
-            listObj <- callJMethod(sdf, "collect")
+          function(rdd) {
+            rddIn <- toRDD(rdd)
```

Comment:
I agree. The only reason it's `rdd` now is because the RDD version of `collect` defines the generic with an argument called `rdd` specifically. This is actually going to likely be an ongoing problem if we continue to have overlap between DataFrame and RDD methods so it'd be good if we figured out a good method for handling this problem in general.